### DISCUSSION TITLE
Monitor crashes after finite amount of DDC commands with the Apple USB-C to HDMI multiport adapter

### DISCUSSION THREAD

**Discussion by desmondw** [MonitorControl/MonitorControl#836] [General]
### Before opening the issue, have you...?

- [X] Searched for existing issues
- [X] Read through [the Monitor-Troubleshooting Wiki](https://github.com/MonitorControl/MonitorControl/wiki/Monitor-Troubleshooting)
- [X] Updated MonitorControl to the latest version (if applicable)

### Describe the issue

Both the brightness and volume slider initially work but then the monitor connection crashes. To reproduce:

1. Increase or decrease the volume or brightness slider through either the keyboard or the UI.
2. Note that initially it works as expected
3. Continue to slide value up and down a couple times and notice that the OSD still reflects your input, but that the brightness or volume has stopped changing
4. Wait a couple of seconds for the connection to the monitor to die (which requires unplugging and plugging back in to reconnect)

Note that changing the values successfully and then waiting does not instigate the issue. It instead appears to occur after a specific number of inputs or transition amounts.

For example, I lowered the brightness with the UI slider, waited a minute, then adjusted with the brightness via keyboard before seeing the issue appear again. In this case it appeared to require less keyboard inputs, suggesting a shared input/transition cap before the crash occurs.

### Expected behavior

Not crash the monitor

### Anything else?

I was able to use [ddcctl](https://github.com/kfix/ddcctl) without any issue, repeatedly toggling brightness between 10 and 90 - e.g. `ddcctl -d 1 -b 10`

- Monitor Cable(s)/Connection(s): MacBook Pro (16-inch, 2019) -> [USB-C Digital AV Multiport Adapter](https://www.apple.com/shop/product/MUF82AM/A/usb-c-digital-av-multiport-adapter) -> [4k HDMI 2.1 cable](https://www.amazon.com/gp/product/B073RBWVBL) -> [Aorus CV27Q](https://www.amazon.com/gp/product/B07X3Y9VTS/)
- [Video of bug](https://drive.google.com/file/d/11QkJJU2fOpaYt7Kc2qNJmbJhR26y2_Si/view?usp=sharing)
- Monitor specs ![monitor](https://user-images.githubusercontent.com/1052705/145084854-d99c2a8d-2fa1-4a53-90be-ef712cbb685a.jpg)

### Environment Information (please complete the following information)

```markdown
- macOS version: 12.0.1 Monterey
- Mac model: MacBook Pro (16-inch, 2019)
- MonitorControl version: v4.0.2
- Monitor(s): Aorus CV27Q
- Monitor Cable(s)/Connection(s): MacBook Pro (16-inch, 2019) -> USB-C Digital AV Multiport Adapter -> 4k HDMI 2.1 cable -> Aorus CV27Q
- Apple Silicon/M1 (yes or no): no
```

**Comment by waydabber** [comment]
Hi, you might want to try disabling `Enable smooth brightness transitions` as that will severely decrease the number of commands sent to the display, and avoid using the slider, use instead the keyboard, or if you use the slider, do not scroll it but point and click to a direct value. If the display is having a firmware problem with handling too many ddc requests (possibly a register is overflowing inside), this might help.

**Reply by desmondw** [reply]
I did also try disabling `Enable smooth brightness transitions`, but again saw no difference. I had also tried slowing down inputs but saw no difference.

As you suggested, I just now tested alternating the brightness between near 0 and near 100 by clicking on the slider. It took slightly longer, but got to the same result. Considering this, I decided to see if I could narrow down if it's an issue of input volume, timing, or both.

It appears that through any combination of keyboard controls (volume up, volume down, volume mute, brightness down, brightness up), _it took exactly 39 key presses to cause the crashed state_. It did not matter if I put the proceeding key presses in fast or slow. In one instance I put in 38 key presses, waited 5 minutes, saw no issue, pressed a key once more, and the issue presented.

**Reply by waydabber** [reply]
It would be interesting to try Lunar (@alin23) and see if that induces the same issue as it also has sliders and keyboard control. You might also want to try what happens if you connect your mac directly to the display - as it might be possible that the multiport adapter has some kind of firmware issue (but it is not very likely).

**Comment by desmondw** [comment]
I tested with Lunar and it also crashes the connection after some usage.

Additionally, I've done some further work with [ddctl](https://github.com/kfix/ddcctl) and after running this script:
```
#!/bin/bash
counter=1
while [ $counter -le 100 ]
do
  ddcctl -d 1 -b $counter
  echo $counter
  ((counter++))
done
echo All done
```

found that it would crash the monitor in the mid-50's:
```
D: CGDisplay AED98375-9ED3-BE28-5E18-E58F0956BF56 dispID(#352960775) (2560x1440 0°) 109.00 DPI
I: found 1 external display
I: polling EDID for #1 (ID 352960775 => IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/PEG0@1/IOPP/EGP0@0/IOPP/EGP1@0/IOPP/GFX0@0/AMDRadeonX6000_AmdRadeonControllerNavi14/ATY,Boa@1)
I: got edid.name: AORUS CV27Q
I: got edid.serial: 20310B003166
D: action: b: 57
D: setting VCP control #16 => 57
57
D: CGDisplay AED98375-9ED3-BE28-5E18-E58F0956BF56 dispID(#352960775) (2560x1440 0°) 109.00 DPI
I: found 1 external display
I: polling EDID for #1 (ID 352960775 => IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/PEG0@1/IOPP/EGP0@0/IOPP/EGP1@0/IOPP/GFX0@0/AMDRadeonX6000_AmdRadeonControllerNavi14/ATY,Boa@1)
I: got edid.name: AORUS CV27Q
I: got edid.serial: 20310B003166
D: action: b: 58
D: setting VCP control #16 => 58
E: Failed to send DDC command!
58
D: CGDisplay AED98375-9ED3-BE28-5E18-E58F0956BF56 dispID(#352960775) (2560x1440 0°) 109.00 DPI
I: found 1 external display
I: polling EDID for #1 (ID 352960775 => IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/PEG0@1/IOPP/EGP0@0/IOPP/EGP1@0/IOPP/GFX0@0/AMDRadeonX6000_AmdRadeonControllerNavi14/ATY,Boa@1)
E: Failed to poll display!
59
D: CGDisplay AED98375-9ED3-BE28-5E18-E58F0956BF56 dispID(#352960775) (2560x1440 0°) 109.00 DPI
I: found 1 external display
I: polling EDID for #1 (ID 352960775 => IOService:/AppleACPIPlatformExpert/PCI0@0/AppleACPIPCI/PEG0@1/IOPP/EGP0@0/IOPP/EGP1@0/IOPP/GFX0@0/AMDRadeonX6000_AmdRadeonControllerNavi14/ATY,Boa@1)
E: Failed to poll display!
60
```

I'll report back once I've got a direct HDMI -> USB-C cable to test if the adapter is the issue.

**Reply by waydabber** [reply]
All right. If it crashes with a straightforward cable as well, you might want to reach out to GigaByte about a fox or a firmware update for the display.

**Reply by desmondw** [reply]
Thanks, yeah it seems to be pointing that way isn't it. What do you mean by 'fox'? I couldn't google the term

**Reply by waydabber** [reply]
fox = fix lol. :) sorry!

**Comment by desmondw** [comment]
I further tested with [this cord](https://www.amazon.com/gp/product/B075V5JK36/ref=ppx_yo_dt_b_asin_title_o04_s00?ie=UTF8&th=1) and didn't appear to have any issues.

Testing with a different HDMI-to-HDMI cable with the adapter still produced the error. Using the original setup and testing on all of the usb-c ports on the macbook produced the same error. It does indeed appear to be an issue with the adapter.

I assume that it would be a very common use case for someone to use the primary [usb-c to hdmi adapter](https://www.apple.com/shop/product/MUF82AM/A/usb-c-digital-av-multiport-adapter) that Apple offers on their site. Has this not come up before? Do I have a faulty adapter? Is it just my adapter/monitor combo that is the issue?

**Reply by waydabber** [reply]
It did not come up, but it is strange indeed that the adapter would have this issue. Apple does not officially support DDC in any capacity so probably that's why it did not come up?

It is a good find however. I'll rename the thread so others might know about this issue with the multiport adapter.

**Reply by waydabber** [reply]
But the display crashing is still unexpected so it might be a combination of the multiport adapter and this specific display that causes the issue.

**Reply by jonathanwiley** [reply]
I had similar issues with a different MacBook Pro and monitor, but same adapter.

```
- macOS version: 12.2.1 Monterey
- Mac model: MacBook Pro (14-inch, 2021)
- MonitorControl version: v4.0.2
- Monitor(s): Asus PB278
- Monitor Cable(s)/Connection(s): MacBook Pro (14-inch, 2021) -> USB-C Digital AV Multiport Adapter -> HDMI cable -> Asus PB278
- Apple Silicon/M1 (yes or no): yes
```

Experienced the same behavior using Lunar. Switched out the Apple USB-C Digital AV Multiport Adapter for an old UGreen multiport adapter and no longer have the issue. Amazing app BTW 🙌
